app/src/main/java/com/nervousfish/nervousfish/modules/pairing/APairingHandler.java

```diff
@@ -36,17 +45,25 @@
     APairingHandler(final IServiceLocator serviceLocator) {
         this.serviceLocator = serviceLocator;
         this.database = serviceLocator.getDatabase();
+        this.constants = serviceLocator.getConstants();
     }
 
     /**
-     * {@inheritDoc}
+     * A QOL mehtod that will send a certain list of contacts
+     *
+     * @param contacts A list with contacts
      */
-    @Override
-    public void sendAllContacts() throws IOException {
-        final List<Contact> list = database.getAllContacts();
-        for (final Contact e : list) {
-            sendContact(e);
+    public void sendAllContacts(final Collection<Contact> contacts) throws IOException {
```

Comment:
Should sending the contacts be part of the PAIRING handler? Sending data isn't really a part of the pairing progress, right?